mteb/evaluation/MTEB.py

```diff
@@ -404,21 +475,26 @@ def run(
                     if verbosity >= 1:
                         logger.info(f"Scores: {results}")
 
-                mteb_task_result = MTEBResults.from_task_results(
+                    if task.metadata_dict["name"] not in self.last_evaluated_splits:
+                        self.last_evaluated_splits[task.metadata_dict["name"]] = set()
+                    self.last_evaluated_splits[task.metadata_dict["name"]].add(split)
+
+                new_results = MTEBResults.from_task_results(
                     task,
                     task_results,
                     evaluation_time=evaluation_time,
                     kg_co2_emissions=kg_co2_emissions,
                 )
 
-                # save results
+                if existing_results:
+                    merged_results = self._merge_results(existing_results, new_results)
```

Comment:
probably worth merging using the MTEBResult object, instead of directly on the dict.

```
new_results = MTEBResults(...)
if existing_results:
  new_results.update(existing_results)
```